### PR TITLE
MBL-680 && MBL-682: FeatureFlag client using internally Firebase Remote Config && ANDROID_HIDE_APP_RATING_DIALOG feature flag migrated

### PR DESCRIPTION
**Pull request by Arkariang**
# 📲 What

New `FeatureFlagClient` available, it internally uses `Firebase.remoteConfig`

# 🤔 Why

- Deadline to migrate away from Optimizely quickly approaching, in case more information is needed take a look at this [engineering plan](https://kickstarter.atlassian.net/wiki/spaces/NT/pages/edit-v2/2326298688) with more detail

# 🛠 How

- In a VewModel level the change will be from `optimizely.isFeatureEnabled(OptimizelyFeature.Key.ANDROID_HIDE_APP_RATING_DIALOG)` to `featureFlagClient()?.getBoolean(FlagKey.ANDROID_HIDE_APP_RATING_DIALOG)`, similar on `ThanksViewModelTest`
- The loading strategy is [Load new values for next startup](https://firebase.google.com/docs/remote-config/loading#strategy_3_load_new_values_for_next_startup), meaning on AppLaunch we activate the already present values, and on the OnCompletation call we fetch any update available in the backend. As user it means new values will be applied the next app launch.

# 👀 See
No user journey changes available, but on the debug build. you can check the logs 

--- With feature flag enabled -----
Screenshot 2023-04-05 at 2 00 33 PM

Screenshot 2023-04-05 at 10 03 35 AM

--- With feature flag disabled -----
Screenshot 2023-04-05 at 2 01 35 PM
Screenshot 2023-04-05 at 2 07 46 PM

|  |  |

# 📋 QA

Go to firebase -> Internal Debug if you are running the internal debug build variant -> left menu -> remote config -> Edit the default value for the feature flag `android_hide_app_rating_dialog` -> publish changes -> Launch the app with the internal debug buid variant and filter in logcat by `FeatureFlagClient` -> kill the app an launch again. At this point if you back a project once the `Thank you` page has been presented you should see the line for the feature flag with the value updated

ℹ️  The exchange on the viewModel/ViewModelTest is almost automatic, and no test should be affected, as no busines logic has been changed.

# Story 📖

[MBL-682](https://kickstarter.atlassian.net/browse/MBL-682)
[MBL-680](https://kickstarter.atlassian.net/browse/MBL-680)

[MBL-682]: https://kickstarter.atlassian.net/browse/MBL-682?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ApplicationModule.java [hunk 1]
```diff
@@ -673,7 +694,7 @@ ExperimentsClientType provideOptimizely(final @ApplicationContext @NonNull Conte
     }
 
     final OptimizelyManager optimizelyManager = OptimizelyManager.builder()
-      .withSDKKey(optimizelyEnvironment.getSdkKey())
+      .withSDKKey("777777")
```
Comment (leighdouglas) [inline app/src/main/java/com/kickstarter/ApplicationModule.java:697]: Just curious, what is this change doing?

Comment (Arkariang) [reply app/src/main/java/com/kickstarter/ApplicationModule.java:697]: My bad! I was testing how optimizely behaves with incorrect/onactive credentials, I'll revert that back

**Review by hadia — APPROVED** [review]
😍

**Comment by ycheng-kickstarter** [comment]
LGTM

**Review by ycheng-kickstarter — APPROVED** [review]

**Review by leighdouglas — APPROVED** [review]
👏🏽

**Pull request merged**